core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionServer.scala

```diff
@@ -214,15 +214,15 @@ private[rest] abstract class KillRequestServlet extends RestServlet {
   protected override def doPost(
       request: HttpServletRequest,
       response: HttpServletResponse): Unit = {
-    val submissionId = parseSubmissionId(request.getPathInfo)
-    val responseMessage = submissionId.map(handleKill).getOrElse {
+    val submissionIds = parseSubmissionId(request.getPathInfo)
```

Comment:
I don't think having submission Ids parsed on the request path is a good idea.
I would assume most use cases for batch delete is required when you have a larger number of drivers to delete (otherwise you would be fine just deleting a few one by one).
But most URLs are length limited.
You might be better have creating a new request for this that takes a body